core/src/main/scala/org/apache/spark/scheduler/TaskResult.scala

```diff
@@ -59,7 +59,7 @@ private[spark] class DirectTaskResult[T](
 
     val numUpdates = in.readInt
     if (numUpdates == 0) {
-      accumUpdates = null
+      accumUpdates = Seq()
```

Comment:
yeah, likewise.  I guess in all real spark jobs, the number of accum updates is > 0, which is why this didn't particularly matter except for these mocks.  OTOH if this has any real danger we should just fix this separately (and be sure to include it in 2.0)